### PR TITLE
Enable CI on SM89 

### PR DESCRIPTION
**Pull request by malfet**
Using EC2 G6 instance, based on NVIDIA L4, added to scale config in https://github.com/pytorch/test-infra/pull/5376

To enable more balanced sharding, had to push https://github.com/pytorch/test-infra/commit/148ae19935c9b69971892b2d150f6402e08d179c

Added `@xfailIfSM89` to the following tests:
 - test_fp8_pattern_2
 - test_original_aten_preserved_split_addmm
 - test_sparse_semi_structured_scaled_mm
 - test_sparse_semi_structured_scaled_mm_fp8
 - test_sparse_fp8fp8_mm
 
Increased tolerance to 2e-4 for `RNNTest.BidirectionalMultilayerGRU_CPU_vs_CUDA`

Skipped following inductor tests (that either flaky OOMs or timeouts):
 - test_reduction_fn_std_float64
 - test_reduction_fn_var_mean_float64
 - test_multi_output_unbacked_custom_op

cc: @wdvr 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: test/cpp/api/rnn.cpp [hunk 1]
```diff
@@ -552,6 +555,13 @@ TEST_F(RNNTest, BidirectionalLSTMReverseForward_CUDA) {
 }
 
 TEST_F(RNNTest, BidirectionalMultilayerGRU_CPU_vs_CUDA) {
+#ifdef USE_CUDA
+  // Get device properties
+  const auto prop = at::cuda::getCurrentDeviceProperties();
+  const auto tolerance = prop->major == 8 && prop->minor == 9 ? 2e-5 : 1e-5;
```
Comment (wdvr) [inline test/cpp/api/rnn.cpp:561]: is this really the way to go? so sm8.9 is 2E-5 because of fp16/fp8? Or what is the reason for the drop in accuracy?

**Review by wdvr — APPROVED** [review]

File: test/cpp/api/rnn.cpp (hunk 1 above)
Comment (malfet) [reply test/cpp/api/rnn.cpp:561]: I'm not sure, but IMO it's OK to do for now and investigate cause of the failure later, I'll create an issue to track all skipped/xfailed tests that were added here to be investigated later.

I can skip the test if you think it's a better behavior.

File: test/inductor/test_torchinductor_dynamic_shapes.py [hunk 2]
```diff
@@ -575,6 +581,7 @@ def f(x):
 
         f(torch.tensor([3], device=device))
 
+    @skipCUDAIf(IS_SM89, "Fails(with OOMS) on SM89")
```
Comment (ZainRizvi) [inline test/inductor/test_torchinductor_dynamic_shapes.py:584]: nit: Why not expand the newly added `xFailIfSM89` to accept an optional message, so that you can use the same skip function across the board?

File: test/cpp/api/rnn.cpp (hunk 1 above)
Comment (ZainRizvi) [reply test/cpp/api/rnn.cpp:561]: A comment explaining the different tolerance values could be helpful to future maintainers here

**Review by ZainRizvi — APPROVED** [review]

File: test/cpp/api/rnn.cpp (hunk 1 above)
Comment (wdvr) [reply test/cpp/api/rnn.cpp:561]: No preference, we could also just change to 2e-5 across all hardware to simplify.

**Review by wdvr — APPROVED** [review]

File: test/cpp/api/rnn.cpp (hunk 1 above)
Comment (wdvr) [reply test/cpp/api/rnn.cpp:561]: Actually, seems to fail now with outcome 4E-5

File: test/inductor/test_torchinductor_dynamic_shapes.py (hunk 2 above)
Comment (malfet) [reply test/inductor/test_torchinductor_dynamic_shapes.py:584]: Because when it OOMs, it can not `xfail`

File: test/cpp/api/rnn.cpp (hunk 1 above)
Comment (malfet) [reply test/cpp/api/rnn.cpp]: Yeah, I've increased it to 2e-4 and created issue to track those failures
